docs/modules/ROOT/pages/index.adoc

```diff
@@ -271,11 +279,12 @@ It uses transaction functions of the driver:
 @POST
 public CompletionStage<Response> create(Fruit fruit) {
     AsyncSession session = driver.asyncSession();
-    return session
+    CompletionStage<Record> cs = session
         .writeTransactionAsync(tx -> tx
             .runAsync("CREATE (f:Fruit {name: $name}) RETURN f", Values.parameters("name", fruit.name))
             .thenCompose(fn -> fn.singleAsync())
         )
+    return threadContext.withContextCapture(cs)
```

Comment:
@cescoffier I think this change warrants an explanation but it's above my understanding of the reactive stuff.
Could you clarify here so that @michael-simons can add a callout.

Reply:
Yes, sure.

It's related to Context Propagation with Completion Stage. Completion Stage, unlike Mutiny, does not have the hook to automatically "capture and restore" the context. So, we need to use this construct.

So, instead of returning the "bare" CompletionStage (cs), it captures the current context, and will restore it before calling a continuation method of the cs (thenSomething, such as thenCompose).  So, in the callback, the context is restored and so available.

Reply:
Would it make sense to change the docs than to use the drivers reactive api, too? I’ll can have a stab at it on Monday. 

The reactive api exposes reactive streams (publisher and friends).

Reply:
The last time I checked it was not great. Basically, they are creating Publisher on top of CompletionStage. So, for now, let's not use it. @jponge did a much deeper analysis.

Reply:
Yep, this is accurate.